File: cs/demo/Ice/value/InitialI.cs
// **********************************************************************
//
// Copyright (c) 2003-2013 ZeroC, Inc. All rights reserved.
//
// This copy of Ice is licensed to you under the terms described in the
// ICE_LICENSE file included in this distribution.
//
// **********************************************************************

using Demo;

class InitialI : InitialDisp_
{
    internal InitialI(Ice.ObjectAdapter adapter)
    {
        _simple = new Simple();
        _simple.message = "a message 4 u";
        
        _printer = new PrinterI();
        _printer.message = "Ice rulez!";
        _printerProxy = PrinterPrxHelper.uncheckedCast(adapter.addWithUUID(_printer));
        
        _derivedPrinter = new DerivedPrinterI();
        _derivedPrinter.message = _printer.message;
        _derivedPrinter.derivedMessage = "a derived message 4 u";
        adapter.addWithUUID(_derivedPrinter);
    }
    
    public override Simple getSimple(Ice.Current current)
    {
        return _simple;
    }
    
    public override void getPrinter(out Printer impl, out PrinterPrx proxy, Ice.Current current)
    {
        impl = _printer;
        proxy = _printerProxy;
    }
    
    public override Printer getDerivedPrinter(Ice.Current current)
    {
        return _derivedPrinter;
    }
    
    public override Printer updatePrinterMessage(Printer printer, Ice.Current current)
    {
	printer.message = "a modified message 4 u";
        return printer;
    }

    public override void throwDerivedPrinter(Ice.Current current)
    {
        DerivedPrinterException ex = new DerivedPrinterException();
        ex.derived = _derivedPrinter;
        throw ex;
    }
    
    public override void shutdown(Ice.Current current)
    {
        current.adapter.getCommunicator().shutdown();
    }
    
    private Simple _simple;
    private Printer _printer;
    private PrinterPrx _printerProxy;
    private DerivedPrinter _derivedPrinter;
}

File: vb/demo/Ice/value/InitialI.vb
' **********************************************************************
'
' Copyright (c) 2003-2014 ZeroC, Inc. All rights reserved.
'
' This copy of Ice is licensed to you under the terms described in the
' ICE_LICENSE file included in this distribution.
'
' **********************************************************************

Imports Demo

Public Class InitialI
    Inherits InitialDisp_

    Public Sub New(ByVal adapter As Ice.ObjectAdapter)
        _simple = New Simple
        _simple.message = "a message 4 u"

        _printer = New PrinterI
        _printer.message = "Ice rulez!"
        _printerProxy = PrinterPrxHelper.uncheckedCast(adapter.addWithUUID(_printer))

        _derivedPrinter = New DerivedPrinterI
        _derivedPrinter.message = _printer.message
        _derivedPrinter.derivedMessage = "a derived message 4 u"
        adapter.addWithUUID(_derivedPrinter)
    End Sub

    Public Overloads Overrides Function getSimple(ByVal current As Ice.Current) As Simple
        Return _simple
    End Function

    Public Overloads Overrides Sub getPrinter(ByRef impl As Printer, ByRef proxy As PrinterPrx, ByVal current As Ice.Current)
        impl = _printer
        proxy = _printerProxy
    End Sub

    Public Overloads Overrides Function getDerivedPrinter(ByVal current As Ice.Current) As Printer
        Return _derivedPrinter
    End Function

    Public Overloads Overrides Function updatePrinterMessage(ByVal printer As Printer, ByVal current As Ice.Current) As Printer
	printer.message = "a modified message 4 u"
        Return printer
    End Function

    Public Overloads Overrides Sub throwDerivedPrinter(ByVal current As Ice.Current)
        Dim ex As DerivedPrinterException = New DerivedPrinterException
        ex.derived = _derivedPrinter
        Throw ex
    End Sub

    Public Overloads Overrides Sub shutdown(ByVal current As Ice.Current)
        current.adapter.getCommunicator().shutdown()
    End Sub

    Private _simple As Simple
    Private _printer As Printer
    Private _printerProxy As PrinterPrx
    Private _derivedPrinter As DerivedPrinter

End Class
